src/gpg-list-options.c

```diff
@@ -80,6 +80,8 @@ static size_t read_option(uint8_t *untrusted_option) {
         case '\0':
             return i;
         case 'A'...'Z': // GnuPG is case-insensitive, but this code is case-sensitive; casefold
+            if (!is_client)
+                errx(1, "Uppercase letter %c should have been casefolded on the client side", untrusted_option[i]);
```

Comment:
This will break if somebody upgrades server but not the client. Since there is no significant value from this tightening (instead of just case folding on the server side too), please drop.